src/mpl/include/mpl_thread_priv.h

```diff
@@ -10,7 +10,7 @@
 extern int MPL_is_threaded;
 void MPL_set_threaded(int isThreaded);
 
-#if MPL_THREAD_PACKAGE_NAME != MPL_THREAD_PACKAGE_NONE && !defined(MPL_TLS)
+#if !defined(MPL_TLS)
```

Comment:
I believe the logic is changed here.
When  `MPL_THREAD_PACKAGE_NAME != MPL_THREAD_PACKAGE_NONE` holds,  the `threadpriv` version is chosen in the current one, though neither is activated previously.

Reply:
I misunderstand something critical. It looks okay to me.